driver-async/src/test/unit/com/mongodb/async/client/FlatteningSingleResultCallbackSubscriptionSpecification.groovy

```diff
@@ -64,12 +68,17 @@ class FlatteningSingleResultCallbackSubscriptionSpecification extends Specificat
         observer.assertNoTerminalEvent()
 
         when:
-        listSingleResultCallback.onResult([1, 2, 3, 4], null)
+        1000.times { executor.submit { observer.requestMore(1) } }
```

Comment:
This doesn't guarantee the race condition was always hit - but over time before the fix it would cause these tests to fail.